src/main/java/ru/yandex/qatools/allure/jenkins/AllureReportBuildAction.java

```diff
@@ -222,24 +227,95 @@ public String getBuildUrl() {
         return run.getUrl();
     }
 
-    @SuppressWarnings("unused")
-    public ArchiveReportBrowser doDynamic(final StaplerRequest req,
-                                          final StaplerResponse rsp)
-            throws IOException, ServletException, InterruptedException {
-        final FilePath archive = new FilePath(run.getRootDir()).child("archive/allure-report.zip");
-        final ArchiveReportBrowser archiveReportBrowser = new ArchiveReportBrowser(archive);
-        archiveReportBrowser.setReportPath(this.getReportPath());
-        return archiveReportBrowser;
+    @Override
+    public void onAttached(final Run<?, ?> attachedRun) {
+        this.run = attachedRun;
     }
 
     @Override
-    public void onAttached(final Run<?, ?> r) {
-        run = r;
+    public void onLoad(final Run<?, ?> loadedRun) {
+        this.run = loadedRun;
     }
 
-    @Override
-    public void onLoad(final Run<?, ?> r) {
-        run = r;
+    @SuppressWarnings("unused")
+    public Object doDynamic(final StaplerRequest request, final StaplerResponse response)
+        throws IOException, InterruptedException {
+
+        final FilePath runRootDir = new FilePath(run.getRootDir());
+        final String reportDirName = getReportPath();
+
+        final FilePath reportDirectoryUnderBuild = runRootDir.child(reportDirName);
+        if (reportDirectoryUnderBuild.exists()) {
+            return new DirectoryReportBrowser(reportDirectoryUnderBuild);
```

Comment:
I strongly recommend this class be removed in favor of Jenkins's built-in `DirectoryBrowserSupport`. That's how Jenkins serves workspaces, archived artifacts, etc.

Whatever is happening here seems very unsafe at a glance.